test/asset-swapper/utils/market_operations_utils/path_test.ts

```diff
@@ -93,6 +97,52 @@ describe('Path', () => {
         });
     });
 
+    describe('hasTwoHop()', () => {
+        it('Returns false when the path does not include a two hop', () => {
+            const path = Path.create(
+                {
+                    side: MarketOperation.Sell,
+                    inputToken: 'fake-input-token',
+                    outputToken: 'fake-output-token',
+                },
+                [
+                    createFakeFillWithFlags(SOURCE_FLAGS[ERC20BridgeSource.UniswapV3]),
+                    createFakeFillWithFlags(SOURCE_FLAGS[ERC20BridgeSource.Curve]),
+                ],
+                ONE_ETHER,
+                {
+                    inputAmountPerEth: new BigNumber(1),
+                    outputAmountPerEth: new BigNumber(1),
+                    exchangeProxyOverhead: () => new BigNumber(0),
+                },
+            );
+
+            expect(path.hasTwoHop()).to.be.false();
+        });
+
+        it('Returns true when the path includes a two hop', () => {
+            const path = Path.create(
+                {
+                    side: MarketOperation.Sell,
+                    inputToken: 'fake-input-token',
+                    outputToken: 'fake-output-token',
+                },
+                [
+                    createFakeFillWithFlags(SOURCE_FLAGS[ERC20BridgeSource.UniswapV3]),
+                    createFakeFillWithFlags(SOURCE_FLAGS[ERC20BridgeSource.MultiHop]),
```

Comment:
does this mean this path has 3 hops? 1. UniswapV3 and 2/3. Multihop? i'm just a little confused by the meaning of the presence of the MultiHop flag.

Reply:
It means that the path has:
<img width="837" alt="image" src="https://user-images.githubusercontent.com/9102479/210426718-5bcebd64-bb7c-40d6-878d-4c6f2b4ce93e.png">.

The proportion depends on the fill amount. For this test case, it's only testing flag so I omitted the fill amount intentionally. 

(I think a path is a misnomer -- probably Route makes more sense... I will consider renaming it in the future).

Flag is essentially a set of DEXes used in the path.